packages/script-editor/src/ScriptEditor.tsx

```diff
@@ -72,14 +71,14 @@ export class ScriptEditor extends DocumentWidget<
   DocumentRegistry.ICodeModel
 > {
   private runner: ScriptRunner;
-  private kernelName: string;
-  private dockPanel: DockPanelSvg;
-  private outputAreaWidget: OutputArea;
-  private scrollingWidget: ScrollingWidget<OutputArea>;
+  private kernelName?: string;
+  private dockPanel?: DockPanelSvg;
+  private outputAreaWidget?: OutputArea;
+  private scrollingWidget?: ScrollingWidget<OutputArea>;
   private model: any;
   private emptyOutput: boolean;
   private runDisabled: boolean;
-  private kernelSelectorRef: RefObject<ISelect>;
+  private kernelSelectorRef: RefObject<ISelect> | null;
```

Comment:
Why are you using `... | null` rather than `kernelSelectorRef?: RefObject...`? (I don't have a preference, just wondering the difference)

Reply:
(seeing this a few other places in the code)

Reply:
`xxx?: yyy` is equivalent to `xxx: yyy | undefined` which isn't the same as `xxx: yyy | null`. I prefer `xxx?: yyy`, but we use `null` in a lot of places and I didn't want to change that and accidentally break things